Content.Server/GameObjects/Components/AdvertiseComponent.cs

```diff
@@ -0,0 +1,140 @@
+
+            // Load advertisements pack
+            if (string.IsNullOrEmpty(PackPrototypeId) || packPrototype == null)
+            {
+                // If there is no pack, log a warning and remove the component
+                Logger.Warning($"{Owner} has {Name}Component but no advertisments pack.");
+                Owner.RemoveComponent<AdvertiseComponent>();
+                return;
+            }
+
+            _advertisements = packPrototype.Advertisements;
+
+            // Do not start timer if advertisement list is empty
+            if (_advertisements.Count == 0)
+            {
+                // If no advertisements could be loaded, log a warning and remove component
+                Logger.Warning($"{Owner} tried to load advertisements pack without ads.");
+                Owner.RemoveComponent<AdvertiseComponent>();
+                return;
+            }
+
+            // Throw exception if MinWait is smaller than 1.
+            if (MinWait < 1)
+            {
+                throw new PrototypeLoadException($"{Owner} has illegal minWait for {Name}Component: {MinWait}.");
+            }
+
+            // Throw exception if MinWait larger than MaxWait.
+            if (MinWait > MaxWait)
+            {
+                throw new PrototypeLoadException($"{Owner} should have minWait greater than or equal to maxWait for {Name}Component.");
+            }
+
+            // Start timer at initialization
+            RefreshTimer();
+        }
+
+        public override void OnRemove()
+        {
+            _cancellationSource.Cancel();
+            _cancellationSource.Dispose();
+
+            base.OnRemove();
+        }
+
+        /// <summary>
+        /// Say advertisement and restart timer.
+        /// </summary>
+        private void SayAndRefresh()
+        {
+            // Say advertisement
+            _chatManager.EntitySay(Owner, Loc.GetString(_random.Pick(_advertisements)));
+
+            // Refresh timer to repeat cycle
+            RefreshTimer();
+        }
+
+        /// <summary>
+        /// Refresh cancellation token and spawn new timer with random wait between <see cref="MinWait"/>
+        /// and <see cref="MaxWait"/>.
+        /// </summary>
+        private void RefreshTimer()
+        {
+            // Generate new source
+            _cancellationSource.Cancel();
+            _cancellationSource.Dispose();
+            _cancellationSource = new CancellationTokenSource();
+
+            // Generate random wait time, then create timer
+            var wait = _random.Next(MinWait * 1000, MaxWait * 1000);
+            Owner.SpawnTimer(wait, SayAndRefresh, _cancellationSource.Token);
+        }
+
+        /// <summary>
+        /// Pause the advertising until <see cref="Resume"/> is called.
+        /// </summary>
+        public void Pause()
+        {
+            // Cancel current timer
+            _cancellationSource.Cancel();
+        }
```

Comment:
Technically it's more of a "Stop" isn't it?

Reply:
For the person using the component it's ultimatelly the same as pausing and resuming, and I feel like it makes more sense this way. Can change it if you want though.